### PR TITLE
docs: add match stream route with SNI tutorial

### PR DESCRIPTION
**Pull request by cr7258**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Please check the lint error

**Comment by cr7258** [comment]
> 

D

**Comment by cr7258** [comment]
I fix the lint error.

**Comment by tao12345666333** [comment]
Thanks

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
@navendu-pottekkat PTAL, Thanks!

File: docs/en/latest/tutorials/match-stream-route-by-sni.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+Instead of requiring a different IP address for each SSL site, you can use SNI to install and configure multiple SSL sites to one IP address.
+
+This guide walks through how to use the ApisixTls and ApisixRoute to route TLS-encrypted traffic to the TCP-based services with SNI.
+
+## Prerequisites
+
+- an available Kubernetes cluster.
+- an available APISIX and APISIX Ingress Controller installation.
+
+First of all, when installing APISIX, we need to enable TLS for the TCP address for APISIX in the Helm Chart, assume that enable tls on TCP port 6379.
```
Comment (shreemaan-abhishek) [inline docs/en/latest/tutorials/match-stream-route-by-sni.md:34]:
```suggestion
First of all, when installing APISIX, we need to enable TLS for the TCP address for APISIX in the Helm Chart, assume that TLS is enabled on TCP port 6379.
```

File: docs/en/latest/tutorials/match-stream-route-by-sni.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  name: redis-2-tls
+spec:
+  hosts:
+    - redis-2.com
+  secret:
+    name: redis-2-secret
+    namespace: default
+```
+
+## Create ApisixRoute match the stream route with SNI
```
Comment (shreemaan-abhishek) [inline docs/en/latest/tutorials/match-stream-route-by-sni.md:169]:
```suggestion
## Create ApisixRoute that matches the stream route with SNI
```

**Review by navendu-pottekkat — APPROVED** [review]

**Comment by navendu-pottekkat** [comment]
Will update and index in a different PR as the "tutorials" section is being updated.

**Pull request merged**